apps/ejabberd/src/ejabberd_receiver.erl

```diff
@@ -255,13 +259,11 @@ handle_info(_Info, State) ->
 %% The return value is ignored.
 %%--------------------------------------------------------------------
 terminate(_Reason, #state{parser = Parser,
-			  c2s_pid = C2SPid} = State) ->
+        c2s_pid = C2SPid} = State) ->
     free_parser(Parser),
-    if
-        C2SPid /= undefined ->
-            gen_fsm:send_event(C2SPid, closed);
-        true ->
-            ok
+    case C2SPid /= undefined of
```

Comment:
How about we match if C2SPid is undefined in the case clause? There is no need to compare it to `undefined`.